substrate/client/network/src/litep2p/mod.rs

```diff
@@ -864,7 +864,7 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkBackend<B, H> for Litep2pNetworkBac
 								);
 
 								self.event_streams.send(Event::Dht(
-									DhtEvent::ValuePut(libp2p::kad::RecordKey::new(&key))
+									DhtEvent::ValuePut(sc_network_types::kad::Key::new(&key.to_vec()))
```

Comment:
Won't the following work to avoid copying?
```suggestion
									DhtEvent::ValuePut(sc_network_types::kad::Key::from(key.to_vec()))
```
Also below.

Reply:
Even better: we can implement `From` for `sc_network_types::kad::Key` to convert it directly from litep2p key, and then the line will became:
```
DhtEvent::ValuePut(key.into())
```

Reply:
> Even better: we can implement `From` for `sc_network_types::kad::Key` to convert it directly from litep2p key, and then the line will become:
> 
> ```
> DhtEvent::ValuePut(key.into())
> ```

`From` was already implemented for `sc_network_types::kad::Key` :smiley: so now we have `DhtEvent::ValuePut(key.into())`. I'm guessing the tas in the next issue, will be to remove `libp2p` from `behaviour.rs` and `service.rs` ?

Reply:
I'm not entirely sure we can remove `libp2p` from behavior.rs and service.rs since they are specific to libp2p architecture (they implement the behavior and work with swarm events)